### PR TITLE
Release 1.0.0

### PR DESCRIPTION
**Pull request by trendelkampschroer**
Checklist
* [x ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ x] Reset the build number to `0` (if the version changed)
* [ x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml
Comment (borchero) [inline recipe/meta.yaml:16]: SHAs are still missing for `polars-lts-cpu` and `polars-u64-idx` :smile:

Comment (trendelkampschroer) [reply recipe/meta.yaml:16]: Will gladly add them, if you would be so kind and tell me where I can find them and the corresponding sources?

Comment (borchero) [reply recipe/meta.yaml:16]: Going through the Jinja template, you can simply build the URL yourself and then compute the SHA by downloading. For `polars-lts-cpu`, the URL would be:

```
https://pypi.io/packages/source/p/polars-lts-cpu/polars_lts_cpu-1.0.0.tar.gz
```

and you can compute the SHA via `curl`:

```
curl -L https://pypi.io/packages/source/p/polars-lts-cpu/polars_lts_cpu-1.0.0.tar.gz | sha256sum
```

Comment (trendelkampschroer) [reply recipe/meta.yaml:16]: Thanks a lot.

**Comment by ritchie46** [comment]
Just to be sure. Does this limit to numpy < 2?

It should as we don't support 2.0 yet (and might segfault because it calls the wrong ABI), we restricted the version this release.

**Comment by xhochy** [comment]
> Just to be sure. Does this limit to numpy < 2?
> 
> It should as we don't support 2.0 yet (and might segfault because it calls the wrong ABI), we restricted the version this release.

No, the numpy pin is open.

**Comment by borchero** [comment]
> No, the numpy pin is open.

Let's change it then to `numpy >=1.16.0,<2`? :smile:

**Comment by trendelkampschroer** [comment]
One of the builds failed with
```
error occurred: Failed to find tool. Is `ml64.exe` installed?
```
I'd be glad if someone more knowledgable about the (windows) build system could chime in here. I'll try to do some digging, but my best guess is that this is a missing dependency for the windows build path...

**Comment by borchero** [comment]
@0xbe7a can you help with the Windows build? Linux aarch64 also seems to timeout :/

**Comment by 0xbe7a** [comment]
> @0xbe7a can you help with the Windows build? Linux aarch64 also seems to timeout :/

I think the Windows build should be fixed, but I'm not sure yet what's happening with the aarch64-linux builds. It seems like it's not actually timing out (we have a 6-hour timeout for Azure Pipelines) but rather starving the worker process (?).

**Comment by xhochy** [comment]
The `aarch64` builds look like an out-of-memory during linking.

**Comment by trendelkampschroer** [comment]
Thanks a lot everyone, especially @0xbe7a, for  all the help and for fixing the build. Once the reviews are in I'd be happy to merge this.

File: recipe/build.sh
Comment (pavelzw) [inline recipe/build.sh:55]: I think this will result in ml64.exe being Part of our conda package 🤔 either remove this file afterwards or put it somewhere else on PATH (like `$BUILD_PREFIX/bin`)

Comment (trendelkampschroer) [reply recipe/build.sh]: I changed `PREFIX` -> `BUILD_PREFIX` now.

File: recipe/build.sh
Comment (0xbe7a) [inline recipe/build.sh:54]:
```suggestion
```

**Review by pavelzw — APPROVED** [review]

File: recipe/meta.yaml
Comment (xhochy) [inline recipe/meta.yaml:45]:
```suggestion
    - maturin >=1.3.2,<2
```

Comment (trendelkampschroer) [reply recipe/meta.yaml:45]: This was already updated, but got reverted in https://github.com/conda-forge/polars-feedstock/pull/240/commits/020bce2dfdceecdce48c89a465ebb16ff64d6869, @borchero was this intended?

**Comment by ritchie46** [comment]
FYI. Next Polars release we will have numpy 2.0 support. Just to keep you guys busy. ;)

File: recipe/meta.yaml
Comment (trendelkampschroer) [reply recipe/meta.yaml:45]: Once the remaining CI jobs are completed - I don't expect any failures - I'd go ahead and merge this as is, if there are no objections. I can bump the `maturin` dependency in a new PR. This way we can avoid having to wait for another round of CI jobs until this version becomes available via conda-forge. 

Please let me know, if that works for everyone.

Comment (0xbe7a) [reply recipe/meta.yaml:45]: When the CI is through, I'd be fine with bumping the pin and merging without waiting for another whole round of builds.

Comment (trendelkampschroer) [reply recipe/meta.yaml:45]: Ok, then I will bump on this branch, but will not wait for the new round of CI to finish. I'll just merge this after the push.

**Comment by trendelkampschroer** [comment]
> FYI. Next Polars release we will have numpy 2.0 support. Just to keep you guys busy. ;)

Looking forward to getting full `numpy-2.0` support. Great job on the `1.0.0` release 👍 .

Hopefully, the next release will just need to unpin `numpy` and be ready to merge after a single round of CI, then.

File: recipe/meta.yaml
Comment (borchero) [reply recipe/meta.yaml]: I don't know why that disappeared, I didn't (want to) touch it :eyes:

**Comment by trendelkampschroer** [comment]
Thanks a lot @0xbe7a for updating and merging!

**Pull request merged**
